src/main/java/com/sproutsocial/nsq/Connection.java

```diff
@@ -173,7 +173,7 @@ private void sendAuthorization(ServerConfig serverConfig) throws IOException {
             if (client.getAuthSecret() == null) {
                 throw new NSQException("nsqd requires authorization, call client.setAuthSecret before connecting");
             }
-            if (!serverConfig.getTlsV1()) {
+            if (!serverConfig.getTlsV1() && (serverConfig.getWarnWhenNotUsingTls() == null || serverConfig.getWarnWhenNotUsingTls())) {
```

Comment:
Can we just make the config a little-`boolean` that's not-nullable, so we don't need this extra boolean logic?

Reply:
done